pkg/commands/compute/language_toolchain.go

```diff
@@ -18,12 +542,77 @@ func getJsToolchainBinPath(bin string) (string, error) {
 	return strings.TrimSpace(string(path)), nil
 }
 
-func checkJsPackageDependencyExists(bin, name string) bool {
-	// gosec flagged this:
-	// G204 (CWE-78): Subprocess launched with variable
-	// Disabling as the variables come from trusted sources:
-	// The CLI parser enforces supported values via EnumVar.
-	/* #nosec */
-	err := exec.Command(bin, "list", "--json", "--depth", "0", name).Run()
-	return err == nil
+// execCommand opens a sub shell to execute the language build script.
+func execCommand(cmd string, args []string, out, progress io.Writer, verbose bool, timeout int, errlog fsterr.LogInterface) error {
+	s := fstexec.Streaming{
+		Command:  cmd,
+		Args:     args,
+		Env:      os.Environ(),
+		Output:   out,
+		Progress: progress,
+		Verbose:  verbose,
+	}
+	if timeout > 0 {
+		s.Timeout = time.Duration(timeout) * time.Second
+	}
+	if err := s.Exec(); err != nil {
+		errlog.Add(err)
+		return err
+	}
+	return nil
+}
+
+// language enables reducing the number of arguments passed to `build()`.
+//
+// NOTE: We're unable to make the build function generic.
+// The generics support in Go1.18 doesn't include accessing struct fields.
+type language struct {
+	buildScript string
+	buildFn     func(string) (string, []string)
+	errlog      fsterr.LogInterface
+	pkgName     string
+	postBuild   string
+	timeout     int
+}
+
+// build compiles the user's source code into a Wasm binary.
+func build(
```

Comment:
Previously this logic was duplicated _four times_ across the code base. Now the logic is shared across all supported languages.